dmoj/graders/base.py

```diff
@@ -1,15 +1,21 @@
+from dmoj.config import ConfigNode
+from dmoj.problem import BatchedTestCase, TestCase
 from dmoj.utils.unicode import utf8bytes
 
 
 class BaseGrader(object):
-    def __init__(self, judge, problem, language, source):
+    def __init__(self, judge, problem, language, source, meta=None):
         self.source = utf8bytes(source)
         self.language = language
         self.problem = problem
         self.judge = judge
+        self.meta = ConfigNode(meta or {})
         self.binary = self._generate_binary()
         self._terminate_grading = False
         self._current_proc = None
+        self._batch_counter = 0
+        self._testcase_counter = 0
+        self.is_pretested = self.meta.get('pretests-only', False) and 'pretest_test_cases' in self.problem.config
```

Comment:
Huh? We control all the users of this...

Reply:
`is_pretested` is also called from `dmoj/judge.py`